content/reference/wadm/api.md

```diff
@@ -0,0 +1,218 @@
+---
+title: "API Reference"
+date: 2018-12-29T11:02:05+06:00
+weight: 2
+draft: false
+description: "wadm API Specification"
+---
+
+The normal way to interact with a `wadm` installation (which could be a single server or a cluster) is through the [wash](../../wash) command-line tool. However, if you are planning on creating your own integration or writing a non-Rust language binding, then this reference will help.
+
+### ⚠️ Caution
+_wadm and its corresponding API are under active development_. Many components are not yet written at all. This document serves as a means to solicit feedback and collaborate on API design and will likely change multiple times.
+
+## Topic Space
+The wadm API is exposed entirely as a NATS service on a topic space. All of the API operations will occur as _requests_ on a topic in the following format:
+
+```
+wadm.api.{lattice-id}.{noun}.{verb}
+```
+
+All requests and responses on this topic are encoded as JSON.
+
+If a consumer wishes to monitor events in real-time (for example, you want to be notified as soon as compensating actions are taken), then it can subscribe to the following topic:
+
+```
+wadm.evt.{lattice-id}
+```
+
+The events on this topic are JSON-encoded [CloudEvents](https://cloudevents.io/).
+
+**NOTE** that all model _names_ are treated like unique identifiers and must conform to the rules governing NATS topic segments. For example, they cannot contain spaces, commas, unprintable characters, or periods.
+
+## Model Persistence
+The following operations pertain to storing and retrieving models. Persistence of models is explicitly and deliberately separated from deployment and deployment management.
+
+
+### Get Model List
+`wadm.api.{lattice}.model.list`
+
+Retrieves a list of models within the given lattice. The status of the model is an aggregate of the statuses of the traits defined within. Only the newest version of the model is included in this list.
+
+**Request**: `none`
+
+**Response**: 
+```json
+{
+    "name": "..",
+    "version": "..",
+    "description": "..",
+    "deployed": true,
+    "status": "compensating|ready|failed|n/a"
+}
+```
+Undeployed models will not provide meaningful data in the `status` field.
+
+### Get Model Spec
+`wadm.api.{lattice}.model.info.{name}`
+
+Retrieves the specification of the model stored with the given name. 
+
+**Request**: `none` 
+
+**Response**:
+
+JSON serialization of [OAM model/yaml](https://github.com/wasmCloud/wadm/tree/main/oam)
+
+
+
+### Store Models
+`wadm.api.{lattice}.model.put.{name}`
+
+Stores the named model. If the model and version being submitted already exist, the request will be rejected. New versions will be appended to the history according to retention policy. Note that this won't automatically deploy a model, it only affects storage.
+
+**Request**: JSON serialization of OAM model
+
+**Response**:
+```json
+{
+    "result": "error|created|newversion",
```

Comment:
This response should have a "versionNumber" field that gets set on creation or update